### ISSUE TITLE
`unravel_index()` argument name `dims` is not consistent with other functions

### ISSUE THREAD

**Issue by SnoopJeDi** [numpy/numpy#10586]
The [`unravel_index()`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.unravel_index.html) function takes an argument named `dims` that corresponds to:

> The **shape** of the array to use for unraveling `indices`.

It seems like in most other places, this argument would be named `shape`, so I propose to rename it to match that convention, unless there is some special reason for this name to be different.

**Comment by eric-wieser** [comment]
I agree that `shape` is a better name, but we need to not break code that calls `np.unravel_index(inds, dims=(...,))`. The easiest way to do that in the C code is something like:

```C
if (ParseNewArgs(...) < 0) {
    save_current_exception();
    if (ParseOldArgs(...) < 0) {
        restore_exception();
        return NULL;
    }
    print_warning_about_old_argname();
}
```
